common/hyperscaler/rules/model/config_test.go

```diff
@@ -0,0 +1,43 @@
+package model
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestLoad(t *testing.T) {
+	t.Run("should load rules from a file", func(t *testing.T) {
+
+		// given
+		expectedRules := []string{"rule1", "rule2"}
+		content := "rule:\n"
+
+		for _, rule := range expectedRules {
+			content += "- " + rule + "\n"
+		}
+		tmpfile, err := os.CreateTemp("", "test*.yaml")
```

Comment:
Alternatively we could use predefined configs in some `testdata` directory.

Reply:
I prefer this approach.

Reply:
This is obvious since you use it.

Reply:
It is easier to understand what is in the file, and how it looks if you can see it in editor/IDE and not infer it from the code that generates config file.

Reply:
My 3 cents. I'd love to have the input file in the `given` section, then few asserts in the `then` section, for example:
```
// given
input := "
rule:
  - rule1
  - rule2
"
// save the input to a tmp file

// when
_, err = config.Load(tmpfile.Name())

// then
// asserts 
```
in such example I see what is the input and see two asserts and I can compare using my eyes.
When creating the input file is hidden by some lines of code I do not see it directly. On the other hand, the above solution seems to be simpler (not sure, but maybe shorter)?